### PR TITLE
Update network utilities and input handling - New functionality

### PR DESCRIPTION
**Pull request by ftavella**
In this PR:
- Consider edge cases on network comparison (`is_same_network()`) to speed up calculation
- Implement a new function (`count_inputs_by_coherence()`) to count the number of coherent and incoherent inputs in a network
- Functions to classify the coherence of single-edge additions to negative-feedback-only networks
  - `is_negative_feedback_network()`
  - `calculate_node_coherence()`
  - `calculate_loop_length_and_type()`
  - `classify_single_addition()` 
- Function to convert connectivity string to matrix (`connectivity_string_to_matrix()`)

### CODE REVIEW

**Review by biphy — CHANGES_REQUESTED** [review]

File: src/network_utilities.jl [hunk 1]
```diff
@@ -166,4 +174,159 @@ function unique_negative_feedback_networks(nodes::Int64)
...
+"""
+function calculate_node_coherence(node_inputs::AbstractArray)
+    n_positive = sum(node_inputs .== 1)
+    n_negative = sum(node_inputs .== -1)
+
+    incoherent = min(n_positive, n_negative)
+    coherent = (max(n_positive, n_negative) - incoherent) / 2
+    coherent = floor(Int, coherent)
+
+    coherence = DataFrame(coherent = coherent, incoherent = incoherent)
```
[the remark this answers is not in the document]
Comment (ftavella) [reply src/network_utilities.jl:249]: That's a great idea for multiple inputs (more than two). I'm going to specify that this function is only for single additions to a negative-feedback-only network

File: src/network_utilities.jl [hunk 2]
```diff
@@ -166,4 +174,159 @@ function unique_negative_feedback_networks(nodes::Int64)
...
+    loop_length = 1
+    loop_type = "unknown"
+    initial_node = loop_start[1]
+    current_node = loop_start[2]
+    next_node = loop_start[2]
+    # Backtrace the loop
+    iterations = 0
+    while next_node != initial_node
+        current_node = next_node
+        next_node = findall(connectivity[next_node, :] .!= 0)[1]
```
[the remark this answers is not in the document]
Comment (ftavella) [reply src/network_utilities.jl:278]: Perfect! I will remove this function and we will use yours moving forward

File: src/network_utilities.jl (hunk 1 above)
[the remark this answers is not in the document]
Comment (ftavella) [reply src/network_utilities.jl:249]: Thinking this again. I will implement what you suggest so we have it in this PR! I'm going to change the return type to be a string.

File: test/network_utilities_tests.jl
Comment (biphy) [inline test/network_utilities_tests.jl:1]: I will rewrite the deleted tests (the ones testing `calculate_loop_length_and_type`) using the function introduced in PR [#25](https://github.com/YangLab-um/BiologicalOscillations.jl/pull/25).

**Review by biphy — APPROVED** [review]
Nice update!

**Pull request merged**